Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -919,35 +896,10 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
                 }
                 completionHandler?(nil)
             }
-        case .setSelectedServer(let selectedServer):
-            let serverSelectionMethod: NetworkProtectionServerSelectionMethod
-
-            switch selectedServer {
-            case .automatic:
-                serverSelectionMethod = .automatic
-            case .endpoint(let serverName):
-                serverSelectionMethod = .preferredServer(serverName: serverName)
-            }
-
-            Task { @MainActor in
-                if case .connected = connectionStatus {
-                    try? await updateTunnelConfiguration(environment: settings.selectedEnvironment, serverSelectionMethod: serverSelectionMethod, reassert: true)
-                }
-                completionHandler?(nil)
-            }
-        case .setSelectedLocation(let selectedLocation):
-            let serverSelectionMethod: NetworkProtectionServerSelectionMethod
-
-            switch selectedLocation {
-            case .nearest:
-                serverSelectionMethod = .automatic
-            case .location(let location):
-                serverSelectionMethod = .preferredLocation(location)
-            }
-
+        case .setSelectedServer, .setSelectedLocation:
```

Comment:
Note that this is now handling both server selection and location selection cases.